src/aiida/transports/plugins/ssh_fabric.py

```diff
@@ -0,0 +1,73 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Transport plugin for SSH connection using ``fabric``.
+
+This subclasses the ``SshTransport`` plugin to replace the connection configuration using the ``fabric`` package. This
+makes the configuration significantly easier for the user since ``fabric`` automatically guesses connection parameters
+by parsing available SSH configuration files on the system. Since ``fabric`` uses ``paramiko`` under the hood, which is
+the library used by the ``SshTransport``, most of the implementation can be reused. The ``paramiko.SshClient`` is
+exposed on the ``fabric.Connection`` instance and is proxied to the ``_client`` attribute of the class where the
+``SshTransport`` base class expects to find it.
+"""
+import fabric
+
+from aiida.common.exceptions import InvalidOperation
+
+from ..transport import Transport
+from .ssh import SshTransport
+
+__all__ = ('SshFabricTransport',)
+
+
+class SshFabricTransport(SshTransport):
+    """Transport plugin for SSH connections with highly-automated connection configuration.
+
+    The connection is made using the ``fabric`` package which will try to use available SSH configuration files on the
+    system to guess the correct configuration for the connection to the relevant hostname.
+    """
+
+    _valid_auth_options = []
+
+    def __init__(self, *args, **kwargs):
+        """Construct a new instance."""
+        Transport.__init__(self, *args, **kwargs)  # Skip the ``SshTransport`` constructor
+        self._connection = fabric.Connection(self.hostname)
+        self._client = self._connection.client
+        self._sftp = None
+
+    def open(self):
+        """Open the connection."""
+        if not self._connection.is_connected:
```

Comment:
It would indeed be nice to get some more clear feedback, for example if I change my `Host` in the `.ssh/config` and then try to test the computer, I get:

```python
❯ verdi computer test todi --print-traceback
Report: Testing computer<todi> for user<lala@lolo>...
* Opening connection... [FAILED]: Error while trying to connect to the computer
  Full traceback:
  Traceback (most recent call last):
    File "/Users/mbercx/project/core/git/aiida-core/src/aiida/cmdline/commands/cmd_computer.py", line 541, in computer_test
      with transport:
    File "/Users/mbercx/project/core/git/aiida-core/src/aiida/transports/transport.py", line 128, in __enter__
      self.open()
    File "/Users/mbercx/project/core/git/aiida-core/src/aiida/transports/plugins/ssh_fabric.py", line 47, in open
      self._sftp = self._connection.sftp()  # This opens the fabric connection ``self._connection`` as well
    File "/Users/mbercx/.aiida_venvs/core/lib/python3.10/site-packages/decorator.py", line 232, in fun
      return caller(func, *(extras + args), **kw)
    File "/Users/mbercx/.aiida_venvs/core/lib/python3.10/site-packages/fabric/connection.py", line 22, in opens
      self.open()
    File "/Users/mbercx/.aiida_venvs/core/lib/python3.10/site-packages/fabric/connection.py", line 665, in open
      result = self.client.connect(**kwargs)
    File "/Users/mbercx/.aiida_venvs/core/lib/python3.10/site-packages/paramiko/client.py", line 349, in connect
      to_try = list(self._families_and_addresses(hostname, port))
    File "/Users/mbercx/.aiida_venvs/core/lib/python3.10/site-packages/paramiko/client.py", line 203, in _families_and_addresses
      addrinfos = socket.getaddrinfo(
    File "/opt/homebrew/Cellar/python@3.10/3.10.14/Frameworks/Python.framework/Versions/3.10/lib/python3.10/socket.py", line 955, in getaddrinfo
      for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
  socket.gaierror: [Errno 8] nodename nor servname provided, or not known
Warning: 1 out of 0 tests failed
```

Same in case I change the name of `~/.ssh/config`.